soccer/gameplay/tactics/wall.py

```diff
@@ -76,84 +79,93 @@ def _add_wall_defenders(self):
         # prioritize the middle elements of the wall
         for i in range(len(pts)):
             pt = pts[i]
+            subbhvr_name = f"robot{i}"
+            if self.has_subbehavior_with_name(subbhvr_name):
+                self.remove_subbehavior(subbhvr_name)
```

Comment:
This may have some unintended consequences. Removing and adding a subbehavior removes the `robot_change_cost` in the subbehaviors role requirements which may cause swapping of roles when they get readded. I'll have to look over the code more to see which is better.

If the wall builds it in such a way where the numbers ascend from one side of the wall to the other, it's better to just add the new one to the end of the list. This way the rest of the wall stays somewhat intact and the new robot moves to the edge of the wall.

If the wall builds it such that robot0 is the center, we should remove all and add all the subbehaviors again since the middle one is the most important.

Reply:
Honestly, based on the `_remove_wall_defenders`, this should be an if statement based on if i is > self.active_defenders, and add the subbehavior, otherwise, just update the pt

Reply:
Ah, I see. I forgot you could update the point directly.